### PR TITLE
Feature: Trakt PIN Auth

### PR DESCRIPTION
**Pull request by Thraxis**
* Implements Trakt PIN Auth as the V2 API will be deprecating the media
center auth method

NOTE: The TRAKT_API_KEY, TRAKT_API_SECRET, and TRAKT_PIN_URL are test
values.  They need to be replaced before final merge

### CODE REVIEW

**Comment by Thraxis** [comment]
Implements https://github.com/SiCKRAGETV/sickrage-issues/issues/1126

File: lib/trakt/trakt.py [hunk 1]
```diff
@@ -1,67 +1,81 @@
...
-    def validateAccount(self):
-        if hasattr(self, 'token'):
-            del(self.token)
+    def traktToken(self, trakt_pin=None, refresh=False, count=0):
+    
+        if count > 3:
+            sickbeard.TRAKT_ACCESS_TOKEN = ''
+            return False
+        elif count > 0:
+            time.sleep(2)
```
Comment (Thraxis) [inline lib/trakt/trakt.py]: Error here.  I forgot to import time.  I will fix this tomorrow

**Comment by Thraxis** [comment]
Rebased to current Develop

**Comment by miigotu** [comment]
@Thraxis what do we need to do in order to get this merged in?

**Comment by fernandog** [comment]
@miigotu we need an official email for sickrage and request TRAKT an official api

**Comment by miigotu** [comment]
@fernandog so lets make a gmail, i think this might help with some of the trakt issues we are having right?

**Comment by Thraxis** [comment]
I know I've been using this for over a month now and haven't seen any issues with it.

**Comment by fernandog** [comment]
@echel0n we need to ask TRAKT an API using sickrage.tv email. can you do that?

**Comment by Thraxis** [comment]
It's not really even asking Trakt itself. You need to:

* Create Trakt.tv account (using sickrage.tv email)
* goto https://trakt.tv/oauth/applications
* Select Create New Application
  Fill in the information for Name, Icon, Description and Redirect URL (use: urn:ietf:wg:oauth:2.0:oob )
  Then click "Save App"
* Replace TRAKT_API_KEY, TRAKT_API_SECRET,  TRAKT_PIN_URL with Client ID, Client Secret and Pin URL in the sickbeard/\_\_init\_\_.py

**Comment by fernandog** [comment]
@Thraxis 

Client ID: 5c65f55e11d48c35385d9e8670615763a605fad28374c8ae55 3a7b7a50651ddd
Client Secret: b53e32045ac122a445ef163e6d859403301ffe9b17fb8321d4 28531b69022a82
PIN URL:
https://trakt.tv/pin/4562

can you please update and test please?

**Comment by Thraxis** [comment]
I'll do that tonight when I get home from work

**Comment by Thraxis** [comment]
@SiCKRAGETV/moderators 

API changed over, PR rebased to current dev, tested out and looks good.

image

For the logo on the API (My Sickrage) I used sickrage/gui/slick/images/ico/favicon-64.png

**Pull request merged**